### DISCUSSION TITLE
Barista Compose

### DISCUSSION THREAD

**Discussion by alorma** [AdevintaSpain/Barista#441] [Ideas]
Discussion around how Barista can contribute to jetpack compose testing.

First of all, I've created a branch `compose_base` with a basic setup of compose in a separate module so we can start exploring how can we build barista around compose

**Comment by alorma** [comment]
First approach: https://github.com/AdevintaSpain/Barista/pull/439

**Reply by costular** [reply]
Nice work @alorma! I think it's a good start. I will be a little bit busy for this and next week but will be able to work on this afterwards 😄

**Comment by costular** [comment]
@alorma Will `barista-compose` be available in a separated artifactory?

**Reply by alorma** [reply]
Yes, it should be available now to download

**Comment by alorma** [comment]
Make this fits on barista compose?

https://kotlinlang.slack.com/archives/CJLTWPH7S/p1633355279467400

```
 composeTestRule.setContent {
     CompositionLocalProvider(LocalContext provides LocalContext.current.createConfigurationContext(Configuration().apply {
         setLocales(LocaleList(Locale.ENGLISH))
     })) {
         //YourUi
     }
 }
```
It avoids having a custom activity or relying on some weird resources modifications

**Comment by NazarDunets** [comment]
Hello there. I've encountered this [issue](https://issuetracker.google.com/issues/235383900) when trying to use `@Repeat`/`@AllowFlaky` with `AndroidComposeTestRule`. Someone in comments proposed a workaround, but I'm not sure how it's supposed to look with barista `FlakyRule`. Currently using `RuleChain` like one in `README.md`

**Reply by Sloy** [reply]
I guess it's something similar. Either play around with order like in the issue, or use a RuleChain to wrap the compose rule inside the flaky rule.

I haven't personally used ComposeRule, so this is just speculation.

**Comment by AShunevych** [comment]
Are there plans to adapt @RetryRule for Jetpack Compose ? For now retry attempt will thrown `java.lang.IllegalStateException: Only a single call to `runTest` can be performed during one test.` or that not Barista problem ?
